lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -719,17 +719,19 @@ RequirementMatch swift::matchWitness(TypeChecker &tc,
     }
   }
 
-  Optional<RequirementEnvironment> reqEnvironment(
-      RequirementEnvironment(dc, reqSig, proto, covariantSelf, conformance));
+  if (!cachedReqEnvironment.hasValue()) {
+    cachedReqEnvironment = RequirementEnvironment(dc, reqSig, proto,
+                                                  covariantSelf, conformance);
```

Comment:
Yeah, so see how we pass 'covariantSelf' to the constructor here. The value of 'covariantSelf' depends on the _witness_; the 'if' statement above only runs if `witness->getDeclContext()->getAsProtocolExtensionContext()`

Reply:
…oh shoot. I missed that.